### ISSUE TITLE
Invisible failure of file logging

### ISSUE THREAD

**Issue by CitrusWire** [godotengine/godot#44815]
**Godot version:**
3.2.3

**OS/device including version:**
Win 7

**Issue description:**
File logging doesn't create logs if the destination directory doesn't exist.
Either:
1) Godot should create the requisite directory structure
2) Issue a Warning level error to let the user know that they need to fix it.

**Steps to reproduce:**
* Enable file logging
* Set log path to: res://logs/godot.log
* Make sure that is to a directory that does *not* exist.
* Run application

**Minimal reproduction project:**
n/a

**Comment by Calinou** [comment]
> Set log path to: res://logs/godot.log

`res://` isn't guaranteed to be writable when the project is exported. It's recommended to keep the default value for the log destination, which is an `user://` data path.

**Comment by CitrusWire** [comment]
@Calinou - Ah, so that's the reason it suggests user://; thanks. The docs do say to use it but not why. I'd suggest this would be worth adding to the doc for log_path.

However, for my purposes I'm far *far* away from exporting.

Further testing:
* user://my_dir/godot.log - creates the subdir; works
* c:\my_dir\godot.log - creates the subdir; works
* res://my_dir/godot.log - does **not** create the subdir

**Comment by Xrayez** [comment]
Somewhat related to #41435, except that logs seem to end up writing to `res://` (opposite to this issue I guess).

**Comment by razonixx** [comment]
Is this taken yet? Sounds like an easy fix.

**Comment by TrickMe** [comment]
I have `flush_stdout_on_print` enabled to check for network errors on a running exported project (runs 24h/7).
The issue I have is that if I run the same project (work in progress) from the editor, the logfiles overwrite each other.

It would be really helpful if there was a way to log files to `res://logs/` on exported projects.
As of now I have to manually change the log directory on export, and it's just a matter of time until I forget and screw up the logs.

Isn't there a way to check if res:// is writeable before writing the logfile?

**Comment by Calinou** [comment]
> Isn't there a way to check if res:// is writeable before writing the logfile on exported projects?

No, but you can set the log path to `res://logs/` in editor builds and `user://logs/` in exported projects by using the `.editor` and `.standalone` [feature tags](https://docs.godotengine.org/en/stable/getting_started/workflow/export/feature_tags.html) respecively.

**Comment by pewcworrell** [comment]
@Calinou I found this old thread and wanted exactly what you mentioned above with using feature tags. I'm working in 4.4. However, when looking [here](https://github.com/godotengine/godot/blob/5abed52fd9d99c4aab7d8c62d505f9ad618e4533/main/main.cpp#L2129), the editor appears to be forbidden from logging. Am I correct? Or am I crazy? This `!editor` check seems like a problem to me. I removed it in our engine and things seem to be working fine.

**Comment by Calinou** [comment]
> However, when looking [here](https://github.com/godotengine/godot/blob/5abed52fd9d99c4aab7d8c62d505f9ad618e4533/main/main.cpp#L2129), the editor appears to be forbidden from logging

The editor itself will never log, but the project running from the editor (i.e. the same editor binary that's running without editor functionality) can still log. `editor` is different from the `TOOLS_ENABLED` compile-time define.
